File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) { 
        // ListNode* List3 = new ListNode(); 
        ListNode* dummy = new ListNode(0);
        ListNode* res = dummy; 


        while(list1 && list2){
            if(list1->val > list2->val){
                res->next = list2; 
                list2 = list2->next; 
            }else{
                res->next = list1 ;
                list1 = list1->next; 
            }
            res = res->next;
        }
        res->next = list1 ? list1 : list2;
        // if (list1 != nullptr){res->next = list1;}
        // if(list2 != nullptr){res->next = list2;}
        ListNode* head = dummy->next; 
        delete dummy; 
        return head; 
        
        
        
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        curr = dummy

        while list1 and list2:
            v1 = list1.val
            v2 = list2.val 

            if (v1 >= v2): 
                curr.next = ListNode(v2)
                list2 = list2.next
            else:
                curr.next = ListNode(v1)
                list1 = list1.next 
            curr = curr.next

        while list1:
            val = list1.val
            curr.next = ListNode(val)
            curr = curr.next 
            list1 = list1.next
        while list2:
            val = list2.val
            curr.next = ListNode(val)
            curr = curr.next
            list2 = list2.next
         
        return dummy.next 
